### PR TITLE
Added CurrentLocaleQueryParamValueProvider #3958

### PR DESCRIPTION
**Pull request by dgodlevskiy**
Fixes: https://github.com/jmix-framework/jmix/issues/3958

### CODE REVIEW

File: jmix-security/security-data/src/main/java/io/jmix/securitydata/impl/CurrentLocaleQueryParamValueProvider.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Component;
+
+/**
+ * Takes query parameter values from the locale of the current authentication.
+ * <p>
+ * Parameter name must be {@code current_locale}
+ */
+@Component("sec_CurrentLocaleQueryParamValueProvider")
+@Order(JmixOrder.HIGHEST_PRECEDENCE + 100)
```
Comment (glebfox) [inline jmix-security/security-data/src/main/java/io/jmix/securitydata/impl/CurrentLocaleQueryParamValueProvider.java:33]: Let's make it `LOWEST_PRECEDENCE - 300`

**Review by glebfox — CHANGES_REQUESTED** [review]

File: jmix-security/security-data/src/main/java/io/jmix/securitydata/impl/CurrentLocaleQueryParamValueProvider.java (hunk 1 above)
Comment (dgodlevskiy) [reply jmix-security/security-data/src/main/java/io/jmix/securitydata/impl/CurrentLocaleQueryParamValueProvider.java]: done

**Review by glebfox — CHANGES_REQUESTED** [review]

File: jmix-security/security-data/src/main/java/io/jmix/securitydata/impl/CurrentLocaleQueryParamValueProvider.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * Parameter name must be {@code current_locale}
+ */
+@Component("sec_CurrentLocaleQueryParamValueProvider")
+@Order(JmixOrder.LOWEST_PRECEDENCE - 300)
+public class CurrentLocaleQueryParamValueProvider implements QueryParamValueProvider {
+
+    public static final String CURRENT_LOCALE_PARAM = "current_locale";
+
+    @Autowired
+    private CurrentAuthentication currentAuthentication;
```
[the remark this answers is not in the document]
Comment (dgodlevskiy) [reply jmix-security/security-data/src/main/java/io/jmix/securitydata/impl/CurrentLocaleQueryParamValueProvider.java]: done

[the remark this answers is not in the document]
Comment (dgodlevskiy) [reply jmix-security/security-data/src/main/java/io/jmix/securitydata/impl/CurrentLocaleQueryParamValueProvider.java]: done

**Review by glebfox — APPROVED** [review]

**Pull request merged**